components/rhs_thread/rhs_thread.tsx

```diff
@@ -159,6 +171,14 @@ export default class RhsThread extends React.Component<Props, State> {
         return false;
     }
 
+    public getSnapshotBeforeUpdate(prevProps: Props, prevState: State) {
```

Comment:
For reference: https://5d4b5feba32acd0008d0df98--reactjs.netlify.app/docs/react-component.html#getsnapshotbeforeupdate

Reply:
Needed to capture scrolled position just prior to posts being added to the thread. React documentation shows a very similar example for this lifecycle method.

Reply:
If we are doing the `isAtBottom` check here i.e `elem.scrollTop + elem.clientHeight === elem.scrollHeight` before adding the posts then we need to tap into this lifecycle otherwise i think we could just skip this and do

```
if (curPostsArray.length > prevPostsArray.length) {
            if (prevState.isAtBottom) {
                this.scrollToBottom();
            }
        }
``` 

Is there a reason why we need to check for in this lifecycle?

Reply:
You're right, that works. I think I was looking at this method first prior to deciding to track the scroll position in `handleScrollStop`.